From 844b7c8b3883690b7f3b641a8368a2e00f9e51c9 Mon Sep 17 00:00:00 2001
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Thu, 9 Feb 2023 17:04:22 -0600
Subject: [PATCH 1/1] linux-aspeed: modify MP2975 VR driver to support MP2862
 VR controller

- MP2975 driver is enabled for build as it's compatible with MP2861/62.
- MP2975 driver is modified to bypass reading the reference offsets.
  It's not supported on MP2862.

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 drivers/hwmon/pmbus/mp2975.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/hwmon/pmbus/mp2975.c b/drivers/hwmon/pmbus/mp2975.c
index 60fbdb371332..6e2785dded60 100644
--- a/drivers/hwmon/pmbus/mp2975.c
+++ b/drivers/hwmon/pmbus/mp2975.c
@@ -627,9 +627,10 @@ mp2975_vout_per_rail_config_get(struct i2c_client *client,
 			return ret;

 		/* Obtain voltage reference offsets. */
-		ret = mp2975_vref_offset_get(client, data, i);
-		if (ret < 0)
-			return ret;
+		/* 'vref_offset' is not supported on MP2862 controller
+		 * it's being used to just check reference offsets,
+		 * there is no side effect when bypass reading reference offsets
+		*/

 		/* Obtain maximum voltage values. */
 		ret = mp2975_vout_max_get(client, data, info, i);
--
2.17.1

